packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,278 @@
+          className: tableColumnInfo[0].className
+        },
+        id: tableColumnInfo[0].id
+      },
+      {
+        title: t("Status"),
+        props: {
+          className: tableColumnInfo[1].className
+        },
+        id: tableColumnInfo[1].id
+      },
+      {
+        title: t("Provider"),
+        props: {
+          className: tableColumnInfo[2].className
+        },
+        id: tableColumnInfo[2].id
+      },
+      {
+        title: t("Buckets"),
+        props: {
+          className: tableColumnInfo[3].className
+        },
+        id: tableColumnInfo[3].id
+      },
+      {
+        title: "",
+        props: {
+          className: tableColumnInfo[4].className
+        },
+        id: tableColumnInfo[4].id
+      }
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null
+  });
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t("Resource Page")}
+      columns={columns}
+      Row={RowRenderer}
+    />
+  );
+};
+
+type CustomData = {
+  launchModal: LaunchModal;
+  resourceModel: K8sModel;
+  kebabActions?: (t: TFunction) => { [key: string]: { value: string } };
+};
+
+const RowRenderer: React.FC<RowProps<K8sResourceCommon, CustomData>> = ({
+  obj,
+  activeColumnIDs,
+  rowData
+}) => {
+  const { resourceModel, launchModal, kebabActions } = rowData;
+  const name = getName(obj);
+  const provider = getProvider(obj);
+  const buckets = getBuckets(obj);
+  return (
+    <>
+      <TableData {...tableColumnInfo[0]} activeColumnIDs={activeColumnIDs}>
+        {name}
+      </TableData>
+      <TableData {...tableColumnInfo[1]} activeColumnIDs={activeColumnIDs}>
+        <OperandStatus operand={obj} />
+      </TableData>
+      <TableData {...tableColumnInfo[2]} activeColumnIDs={activeColumnIDs}>
+        {provider}
+      </TableData>
+      <TableData {...tableColumnInfo[3]} activeColumnIDs={activeColumnIDs}>
+        {buckets + " Buckets"}
+      </TableData>
+      <TableData {...tableColumnInfo[4]} activeColumnIDs={activeColumnIDs}>
+        <Kebab
+          launchModal={launchModal}
+          extraProps={{ resource: obj, resourceModel: resourceModel }}
+          customKebabItems={kebabActions}
+        />
+      </TableData>
+    </>
+  );
+};
+
+type GenericListPageProps = {
+  actions?: ModalMap;
+  resourceModel: K8sModel;
+  kebabActions?: CustomData["kebabActions"];
+};
+
+const GenericListPage: React.FC<GenericListPageProps> = ({
+  actions,
+  resourceModel,
+  kebabActions
+}) => {
+  const { t } = useTranslation();
+  const [Modal, modalProps, launchModal] = useModalLauncher(actions);
+
+  const bucketResource = React.useMemo(
+    () => ({
+      kind: referenceForModel(NooBaaBucketClassModel),
+      namespace: DATA_FEDERATION_NAMESPACE,
+      isList: true
+    }),
+    []
+  );
+  const [bucketResources] =
+    useK8sWatchResource<BucketClassKind[]>(bucketResource);
```

Comment:
why polling again when u r already polling for BucketClass below... when this List component is called...

Reply:
that;s the issue... name of the component is "GenericListPage" now if I use same component for BucketClass (Just like u did here for NSS) then it will poll for same resource twice...

Reply:
moved nss list to data resource list and kept other 2 in generics